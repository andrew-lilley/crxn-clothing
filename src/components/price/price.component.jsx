import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import numeral from "numeral";
import 'numeral/locales/en-gb';
import 'numeral/locales/de';
import { selectActiveCurrencyLangCode } from '../../redux/currency/currency.selectors'

export const Price = ({ price, label, langCode }) => {

  if (langCode !== 'default') {
    numeral.locale(langCode);
  }
  else {
    numeral.reset();
  }

  return (
    <span className='price'>
      { label && `${label}: `}
      {numeral(price).format('$0,0.00')}
    </span>
  )
}

const mapStateToProps = createStructuredSelector({
  langCode: selectActiveCurrencyLangCode
});

export default connect(mapStateToProps)(Price);