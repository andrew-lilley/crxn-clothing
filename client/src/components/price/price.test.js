import React from 'react';
import { shallow } from 'enzyme';
import { Price } from './price.component';

describe('Price component', () => {

  test('should render Price component with no label', () => {
    const mockProps = {
      price: 100,
      label: '',
      langCode: 'en-gb'
    };
    const wrapper = shallow(<Price {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should render Price component with a label', () => {
    const mockProps = {
      price: 100,
      label: 'Total',
      langCode: 'en-gb'
    };
    const wrapper = shallow(<Price {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should render Price component with default langCode', () => {
    const mockProps = {
      price: 100,
      label: '',
      langCode: 'default'
    };
    const wrapper = shallow(<Price {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

});
