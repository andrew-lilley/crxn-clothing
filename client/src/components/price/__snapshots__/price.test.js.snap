// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price component should render Price component with a label 1`] = `
<Fragment>
  Total: 
  £100.00
</Fragment>
`;

exports[`Price component should render Price component with default langCode 1`] = `
<Fragment>
  $100.00
</Fragment>
`;

exports[`Price component should render Price component with no label 1`] = `
<Fragment>
  £100.00
</Fragment>
`;
